lib/rules/no-implied-eval.js

```diff
@@ -28,94 +34,90 @@ module.exports = {
+            return node.type === "MemberExpression" && specifiers.includes(astUtils.getStaticPropertyName(node));
         }
 
         /**
-         * Checks that the parent is a direct descendent of an potential implied eval CallExpression, and if the parent is a CallExpression, that we're the first argument.
-         * @param {ASTNode} node The node to inspect the parent of.
-         * @returns {boolean} Was the parent a direct descendent, and is the child therefore potentially part of a dangerous argument?
-         * @private
+         * Reports if the `CallExpression` node has evaluated argument.
+         * @param {ASTNode} node A CallExpression to check.
+         * @returns {void}
          */
-        function hasImpliedEvalParent(node) {
+        function reportImpliedEvalCallExpression(node) {
+            if (node.type !== "CallExpression") {
+                return;
+            }
 
-            // make sure our parent is marked
-            return node.parent === last(last(impliedEvalAncestorsStack)) &&
+            const [firstArgument] = node.arguments;
 
-                // if our parent is a CallExpression, make sure we're the first argument
-                (node.parent.type !== "CallExpression" || node === node.parent.arguments[0]);
+            if (firstArgument && isEvaluatedString(firstArgument)) {
+                context.report({
+                    node,
+                    messageId: "impliedEval"
+                });
+            }
         }
 
         /**
-         * Checks if our parent is marked as part of an implied eval argument. If
-         * so, collapses the top of impliedEvalAncestorsStack and reports on the
-         * original CallExpression.
-         * @param {ASTNode} node The CallExpression to check.
-         * @returns {boolean} True if the node matches, false if not.
-         * @private
+         * Reports calls of `implied eval` via the global references.
+         * @param {Variable} globalVar A global variable to check.
+         * @returns {void}
          */
-        function checkString(node) {
-            if (hasImpliedEvalParent(node)) {
+        function reportImpliedEvalViaGlobal(globalVar) {
+            const { references, name } = globalVar;
 
-                // remove the entire substack, to avoid duplicate reports
-                const substack = impliedEvalAncestorsStack.pop();
+            references.forEach(ref => {
+                const identifier = ref.identifier;
+                let node = identifier.parent;
 
-                context.report({
-                    node: substack[0],
-                    messageId: "impliedEval"
-                });
-            }
+                while (isSpecifiedMember(node, [name])) {
+                    node = node.parent;
+                }
+
+                if (isSpecifiedMember(node, EVAL_LIKE_FUNCS)) {
+                    node = node.parent;
+                    reportImpliedEvalCallExpression(node);
```

Comment:
Maybe we should check here if the `node` is `callee` of a `CallExpression` (instead of checking if the given node is  `CallExpression` in `reportImpliedEvalCallExpression`).

This currently has a false positive:

```js
/*eslint no-implied-eval: "error"*/
/* eslint-env browser */

foo("", window.setTimeout);
```

Reply:
@mdjermanovic 
Thanks, 👍 I  changed it.